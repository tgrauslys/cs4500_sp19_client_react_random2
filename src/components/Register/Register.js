import React from 'react'

const Register = ({register, updateFirstName, updateLastName, updateUsername, updatePassword, isErrorMessageOn}) => {
    let errorMessage = <div className="row"><div className="col-12">Email already taken! Please choose another email.</div></div>
    if (!isErrorMessageOn) {
        errorMessage = ''
    }
    return (<div className="container">
            <h1>Create your account</h1>
            <br/>
            <div>
                <div className="row">
                    <div className="col-6">
                        <label for="first-name">First name</label>
                        <input id="first-name" className="form-control" onChange={e => updateFirstName(e)}/>
                    </div>
                    <div className="col-6">
                        <label for="last-name">Last name</label>
                        <input id="last-name" className="form-control" onChange={e => updateLastName(e)}/>
                    </div>
                </div>
                <br/>
                <div className="row">
                    <div className="col-12">
                        <label for="email">Email</label>
                        <input id="email" className="form-control" onChange={e => updateUsername(e)}/>
                    </div>
                </div>
                <br/>
                <div className="row">
                    <div className="col-12">
                        <label for="password">Password</label>
                        <input id="password" className="form-control" onChange={e => updatePassword(e)}/>
                    </div>
                </div>
                <br/>
                {errorMessage}
                <div className="row">
                    <div className="col-12">
                    <button onClick={register}
                            style={{margin: '2px'}}
                            type="button" className="btn btn-primary btn-block">Create Account</button>
                    </div>
                </div>
            </div>
        </div>)
}
    

export default Register