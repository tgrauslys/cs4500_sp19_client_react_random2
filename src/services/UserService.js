export default class UserService {
    static instance = null;
    static getInstance() {
        if(UserService.instance === null) {
            UserService.instance = new UserService()
        }
        return this.instance
    }
    findUserById = userId =>
        fetch(`http://localhost:8080/api/users/${userId}`)
            .then(response => response.json())
    findAllUsers = () =>
        fetch("http://localhost:8080/api/users")
            .then(response => response.json())
}
