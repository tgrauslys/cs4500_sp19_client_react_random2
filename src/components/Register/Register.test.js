import registry from '../../data/Register.mock.json'
import React from 'react'
import Register from './Register'
import TestRenderer from 'react-test-renderer'
import UserService from '../../services/UserService'

const userService = UserService.getInstance()
var testUserId

describe('Register Page Test', () => {
    test('Register page renders correctly', () => {
        let updateFirstName = () => {}
        let updateLastName = () => {}
        let updateEmail = () => {}
        let updateUsername = () => {}
        let updatePassword = () => {}
        let submit = () => {}
    
        const testRenderer = TestRenderer.create(
            <Register 
                register={submit}
                updateFirstName={updateFirstName}
                updateLastName={updateLastName}
                updateEmail={updateEmail}
                updateUsername={updateUsername}
                updatePassword={updatePassword}
                isErrorMessageOn={true}/>
        )
    
        let tree = testRenderer.toJSON()
        expect(tree).toMatchSnapshot()
    
        const testInstance = testRenderer.root
    
        const firstNameField = testInstance.findByProps({id: "first-name"})
        const lastNameField = testInstance.findByProps({id: "last-name"})
        const emailField = testInstance.findByProps({id: "email"})
        const usernameField = testInstance.findByProps({id: "username"})
        const passwordField = testInstance.findByProps({id: "password"})
        const registerButton = testInstance.findAllByProps({className: "btn btn-primary btn-block"})
        const fields = testInstance.findAllByProps({className: "form-control"})

        expect(firstNameField).toBeDefined()
        expect(lastNameField).toBeDefined()
        expect(emailField).toBeDefined()
        expect(usernameField).toBeDefined()
        expect(passwordField).toBeDefined()
        expect(registerButton[0]).toBeDefined()
        expect(registerButton[0].type).toBe("button")
        expect(registerButton.length).toBe(1)
        expect(fields.length).toBe(5)
    })
})