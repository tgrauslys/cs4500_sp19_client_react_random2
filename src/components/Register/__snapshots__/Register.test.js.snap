// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Test Register page renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Create your account
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="first-name"
        >
          First name
        </label>
        <input
          className="form-control"
          id="first-name"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-6"
      >
        <label
          for="last-name"
        >
          Last name
        </label>
        <input
          className="form-control"
          id="last-name"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        Email already taken! Please choose another email.
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          onClick={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="button"
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;
